### PR TITLE
Added release steps for publishing helm artifacts in gh-pages

### PR DESCRIPTION
**Pull request by evalsocket**
- Added GitHub workflow for publishing helm artifact in gh-pages. 

Fix: https://github.com/flyteorg/flyte/issues/1111

Tested result : https://github.com/evalsocket/flyte/releases/tag/v0.1.222
Index of registry : https://github.com/evalsocket/flyte/blob/gh-pages/index.yaml
Github Action Test: https://github.com/evalsocket/flyte/runs/3002001581?check_suite_focus=true

Test Helm 
```bash
$ helm repo add flyte https://evalsocket.github.io/flyte
$ helm repo update
$ helm  install sandbox flyte/flyte
```

### CODE REVIEW

**Comment by evalsocket** [comment]
Some notes:
1. https://github.com/helm/charts-repo-actions-demo/blob/main/.github/workflows/release.yaml is the demo of the helm github action but it only works if you run It on every push on master because in that case it creates a new release and publishes helm package. But in our case, we will publish it in release and then it failed (https://github.com/helm/chart-releaser-action/issues?q=No+changes+detected)

File: .goreleaser.yml [hunk 1]
```diff
@@ -11,4 +11,5 @@ release:
   name_template: "Flyte v{{.Version}} milestone release"
   disable: false
   extra_files:
-    - glob: ./release/*
\ No newline at end of file
+    - glob: ./release/*
+    - glob: .cr-release-packages/
```
Comment (EngHabu) [inline .goreleaser.yml:15]: I don't see us copying anything into this... or is that something `cr helm...` expects/populates?

File: script/prepare_artifacts.sh [hunk 2]
```diff
@@ -1,11 +1,31 @@
...
+
+# Download helm chart releaser
+wget -q -O /tmp/chart-releaser.tar.gz https://github.com/helm/chart-releaser/releases/download/v1.2.1/chart-releaser_1.2.1_linux_amd64.tar.gz 
+mkdir -p bin
+tar -xf /tmp/chart-releaser.tar.gz -C bin
+chmod +x bin/cr
+rm /tmp/chart-releaser.tar.gz \
+
+# Clean Git History
+git stash
```
Comment (EngHabu) [inline script/prepare_artifacts.sh:25]: Why is this needed?

File: script/prepare_artifacts.sh [hunk 3]
```diff
@@ -1,11 +1,31 @@
...
+
+# Package helm chart
+bin/cr package helm
+
+# Commit Chart registry to github gh-pages
+bin/cr index --owner flyteorg --git-repo flyte --charts-repo="https://flyteorg.github.io/flyte" --push --token=${FLYTE_BOT_PAT} --release-name-template="{{ .Version }}"
```
Comment (EngHabu) [inline script/prepare_artifacts.sh:31]: Where does it get the `Version` value from? I think they come from here: https://github.com/flyteorg/flyte/blob/master/helm/Chart.yaml#L5-L6 

Will these values be updated as part of this: https://github.com/flyteorg/flyte/actions/workflows/generate-flyte-manifest.yml ?

File: .github/workflows/create-release.yml [hunk 4]
```diff
@@ -16,6 +16,17 @@ jobs:
       - uses: actions/checkout@v2
         with:
           fetch-depth: "0"
+
+      - name: Install Helm
+        uses: azure/setup-helm@v1
+        with:
+          version: v3.4.0
+            
+      - name: Setup Git
```
Comment (EngHabu) [inline .github/workflows/create-release.yml:25]: nit:
```suggestion
      - name: Configure Git
```

**Comment by EngHabu** [comment]
Looks so close! thank you for working on this

File: .goreleaser.yml (hunk 1 above)
Comment (evalsocket) [reply .goreleaser.yml:15]: yes, they populate the tar of helm package

File: script/prepare_artifacts.sh (hunk 2 above)
Comment (evalsocket) [reply script/prepare_artifacts.sh:25]: It can be removed because i added everything in gitignore

File: script/prepare_artifacts.sh (hunk 3 above)
Comment (evalsocket) [reply script/prepare_artifacts.sh:31]: yes, they come from the chart.yaml

Comment (evalsocket) [reply script/prepare_artifacts.sh:31]: Yes, they are coming from chart.yaml and we can't add them in the above workflow because we need to add a version that we will create, and that workflow doesn't know that.


We need to update the value manually

Comment (EngHabu) [reply script/prepare_artifacts.sh:28]: Maybe what we should do is have a placeholder `#VERSION` in the chart.yaml... and we populate that here, temporarily, using the version passed in to the current workflow. We package the chart and push it to the gh-pages branch then we `git stash` that change.... how about that?

Comment (evalsocket) [reply script/prepare_artifacts.sh:28]: yes that will work

Comment (evalsocket) [reply script/prepare_artifacts.sh]: done

**Comment by evalsocket** [comment]
@EngHabu do we need to be generated helm?

**Comment by EngHabu** [comment]
Do you mean `do we need to run make helm in this WF?` or `do we still need make helm at all?`
It runs as part of [this WF](https://github.com/flyteorg/flyte/blob/master/.github/workflows/generate-flyte-manifest.yml#L18) Which I believe is the right thing to do

**Review by EngHabu — APPROVED** [review]
